code/modules/mob/living/carbon/xenomorph/castes/Queen.dm

```diff
@@ -545,9 +545,8 @@
 
 	. += "Pooled Larvae: [stored_larvae]"
 	. += "Leaders: [xeno_leader_num] / [hive?.queen_leader_limit]"
-	if(queen_age_timer_id != TIMER_ID_NULL)
-		var/time_left = time2text(timeleft(queen_age_timer_id) + 1 MINUTES, "mm") // We add a minute so that it basically ceilings the value.
-		. += "Maturity: [time_left == 1? "[time_left] minute" : "[time_left] minutes"] remaining"
+	if(!queen_aged && queen_age_timer_id != TIMER_ID_NULL) // Idk, bug or not, but when timer ends, it leave just "2", this is soft solution to polution
```

Comment:
You could add this to `/mob/living/carbon/xenomorph/queen/proc/make_combat_effective()`

```js
if(queen_age_timer_id != TIMER_ID_NULL)
		deltimer(queen_age_timer_id)
		queen_age_timer_id = TIMER_ID_NULL
```

Would also mean any manual call of that proc would resolve the maturity timer too.

Reply:
This is now error prone where timeleft can now potentially be returning null to pass into time2text.

Use `queen_age_timer_id != TIMER_ID_NULL` or `!queen_aged && queen_age_timer_id != TIMER_ID_NULL)` I don't really care; but I highly recommend what I suggested.